graphql_compiler/compiler/emit_sql.py

```diff
@@ -130,6 +119,17 @@ def _find_columns_used_outside_folds(
         query_path = output_info.location.query_path
         used_columns.setdefault(query_path, set()).add(output_info.location.field)
 
+    # Columns used in the base case of CTE recursions should be made available from parent scope
+    for location, _ in ir.query_metadata_table.registered_locations:
+        if isinstance(location, FoldScopeLocation):
+            continue
+        for recurse_info in ir.query_metadata_table.get_recurse_infos(location):
+            traversal = "{}_{}".format(recurse_info.edge_direction, recurse_info.edge_name)
+            used_columns[location.query_path] = used_columns.get(location.query_path, set()).union(
+                used_columns[location.query_path + (traversal,)]
+            )
+            used_columns[location.query_path].add(edge.from_column)
+
```

Comment:
Moved this code down so that fields used in tags and outputs would be picked up.

Reply:
Awesome